File: 프로그래머스/3/43162. 네트워크/네트워크.py
def solution(n, computers):
    
    def dfs(node):
        visited[node] = True
        for i in range(n):
            if computers[node][i] and not visited[i]:
                dfs(i)
    
    visited = [False] * n
    
    answer = 0
    for i in range(n):
        if not visited[i]:
            dfs(i)
            answer += 1
    

    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    
    function dfs(node){
        visited[node] = true
        for (let i=0; i<n; i++){
            if (!visited[i] && computers[node][i] === 1) {
                dfs(i)
            }
        }
    }
    
    
    visited = Array(n).fill(false)
    
    let answer = 0;
    
    for (let i=0; i<n; i++){
        if (!visited[i]) {
            dfs(i)
            answer += 1
        }
    }
    
    
    
    return answer;
}